### PR TITLE
GLOBAL variables changed to GLOBAL constants

### LINKED ISSUE #1629: refactor airdata
This is a start to have tests for the current solution of airdata processing
https://github.com/Open-MSS/MSS/pull/1627

We should change global variables to global constants and reduce the dependencies so that the parts can functional be tested without so much mocking

### PR DESCRIPTION
**Pull request by rootxrishabh**
fixes #1629

### CODE REVIEW

File: mslib/utils/airdata.py [hunk 1]
```diff
@@ -43,22 +43,42 @@
...
-_airspace_cache = [('ad_asp.xml', '377'), ('ae_asp.xml', '110238'), ('af_asp.xml', '377'), ('ag_asp.xml', '377'),
-                   ('ai_asp.xml', '377'), ('al_asp.xml', '11360'), ('am_asp.xml', '377'), ('ao_asp.xml', '377'),
-                   ('aq_asp.xml', '377'), ('ar_asp.xml', '975402'), ('as_asp.xml', '377'), ('at_asp.xml', '1059077'),
-                   ('au_asp.xml', '9667669'), ('aw_asp.xml', '377'), ('ax_asp.xml', '377'), ('az_asp.xml', '377'),
-                   ('ba_asp.xml', '76717'), ('bb_asp.xml', '377'), ('bd_asp.xml', '377'), ('be_asp.xml', '462667'),
-                   ('bf_asp.xml', '377'), ('bg_asp.xml', '440690'), ('bh_asp.xml', '68519'), ('bi_asp.xml', '377'),
-                   ('bj_asp.xml', '377'), ('bl_asp.xml', '377')
-                 ]
+
+class Airports:
```
Comment (ReimarBauer) [inline mslib/utils/airdata.py]: this is now a 1:1 translation.  Is there anything against trying it with less class definition?

Comment (rootxrishabh) [reply mslib/utils/airdata.py]: I have redefined the class structure of this file.

File: mslib/utils/airdata.py [hunk 2]
```diff
@@ -43,22 +43,30 @@
...
-_airspace_cache = [('ad_asp.xml', '377'), ('ae_asp.xml', '110238'), ('af_asp.xml', '377'), ('ag_asp.xml', '377'),
-                   ('ai_asp.xml', '377'), ('al_asp.xml', '11360'), ('am_asp.xml', '377'), ('ao_asp.xml', '377'),
-                   ('aq_asp.xml', '377'), ('ar_asp.xml', '975402'), ('as_asp.xml', '377'), ('at_asp.xml', '1059077'),
-                   ('au_asp.xml', '9667669'), ('aw_asp.xml', '377'), ('ax_asp.xml', '377'), ('az_asp.xml', '377'),
-                   ('ba_asp.xml', '76717'), ('bb_asp.xml', '377'), ('bd_asp.xml', '377'), ('be_asp.xml', '462667'),
-                   ('bf_asp.xml', '377'), ('bg_asp.xml', '440690'), ('bh_asp.xml', '68519'), ('bi_asp.xml', '377'),
-                   ('bj_asp.xml', '377'), ('bl_asp.xml', '377')
-                 ]
+
+class Global_Constants:
```
Comment (ReimarBauer) [inline mslib/utils/airdata.py]: this is from the name what it is. If you want to describe for what it  is used do add a doc string. This is then also shown in pycharm etc.

As almost all data inside is prefixed by airspace why not naming it like Airspace and reduce the prefixes? 
airspaces = [] may become data = []

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
see comment

**Comment by ReimarBauer** [comment]
Do you see some airdata in the widget?

File: mslib/utils/airdata.py (hunk 2 above)
Comment (rootxrishabh) [reply mslib/utils/airdata.py]: I'll get it done.

**Comment by rootxrishabh** [comment]
> Do you see some airdata in the widget?

Is this correct?
image

File: mslib/utils/airdata.py (hunk 2 above)
Comment (rootxrishabh) [reply mslib/utils/airdata.py]: Should I keep the airports and airports_mtime the same?

**Comment by ReimarBauer** [comment]
For some reason I don't get any airport shown.

They do download and have a big size
```
ls -rtlh downloads/aip/airports.csv 
-rw-rw-r-- 1 reimar reimar 11M Mar 23 12:34 downloads/aip/airports.csv
```

But I always get 
ERROR: Tried to draw airports but none were found. Try redownloading.

Have a look on this line, this misses the new "download" folder. 
https://github.com/rootxrishabh/MSS/blob/i1629/mslib/utils/airdata.py#L126

I assume you have in .config/msui an older airports.csv file  and that's why you can see data. But the newer version from the download folder is not used

**Comment by ReimarBauer** [comment]
is missing
```
diff --git a/mslib/utils/airdata.py b/mslib/utils/airdata.py
index 751d090f..f173f5a9 100644
--- a/mslib/utils/airdata.py
+++ b/mslib/utils/airdata.py
@@ -123,9 +123,9 @@ def get_airports(force_download=False, url=None):
             == QtWidgets.QMessageBox.Yes:
         download_progress(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), url)
 
-    if os.path.exists(os.path.join(OSDIR, "airports.csv")):
-        with open(os.path.join(OSDIR, "airports.csv"), "r", encoding="utf8") as file:
-            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "airports.csv"))
+    if os.path.exists(os.path.join(OSDIR, "downloads", "aip", "airports.csv")):
+        with open(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), "r", encoding="utf8") as file:
+            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "downloads", "aip", "airports.csv"))
             return list(csv.DictReader(file, delimiter=","))
 
     else:
```

**Comment by rootxrishabh** [comment]
> is missing
> 
> ```
> diff --git a/mslib/utils/airdata.py b/mslib/utils/airdata.py
> index 751d090f..f173f5a9 100644
> --- a/mslib/utils/airdata.py
> +++ b/mslib/utils/airdata.py
> @@ -123,9 +123,9 @@ def get_airports(force_download=False, url=None):
>              == QtWidgets.QMessageBox.Yes:
>          download_progress(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), url)
>  
> -    if os.path.exists(os.path.join(OSDIR, "airports.csv")):
> -        with open(os.path.join(OSDIR, "airports.csv"), "r", encoding="utf8") as file:
> -            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "airports.csv"))
> +    if os.path.exists(os.path.join(OSDIR, "downloads", "aip", "airports.csv")):
> +        with open(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), "r", encoding="utf8") as file:
> +            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "downloads", "aip", "airports.csv"))
>              return list(csv.DictReader(file, delimiter=","))
>  
>      else:
> ```

I have synced the current PR branch and I am still not able to reproduce the error, can you please check? Also which 2 branches are you comparing?

**Comment by ReimarBauer** [comment]
> > is missing
> > ```
> > diff --git a/mslib/utils/airdata.py b/mslib/utils/airdata.py
> > index 751d090f..f173f5a9 100644
> > --- a/mslib/utils/airdata.py
> > +++ b/mslib/utils/airdata.py
> > @@ -123,9 +123,9 @@ def get_airports(force_download=False, url=None):
> >              == QtWidgets.QMessageBox.Yes:
> >          download_progress(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), url)
> >  
> > -    if os.path.exists(os.path.join(OSDIR, "airports.csv")):
> > -        with open(os.path.join(OSDIR, "airports.csv"), "r", encoding="utf8") as file:
> > -            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "airports.csv"))
> > +    if os.path.exists(os.path.join(OSDIR, "downloads", "aip", "airports.csv")):
> > +        with open(os.path.join(OSDIR, "downloads", "aip", "airports.csv"), "r", encoding="utf8") as file:
> > +            _airports_mtime = os.path.getmtime(os.path.join(OSDIR, "downloads", "aip", "airports.csv"))
> >              return list(csv.DictReader(file, delimiter=","))
> >  
> >      else:
> > ```
> 
> I have synced the current PR branch and I am still not able to reproduce the error, can you please check? Also which 2 branches are you comparing?


show me ls -l .config/msui

and look at the image, which shows your repository file 

compare

line 125 has the correct path, but those below were not changed.

**Comment by ReimarBauer** [comment]
We not only want to download to os.path.join(OSDIR, "downloads", "aip", "airports.csv") we want also to read from there.

**Comment by rootxrishabh** [comment]
Moin @ReimarBauer, I have made some changes can you please review them?

**Review by ReimarBauer — APPROVED** [review]
Thx

**Pull request merged**